run function is susceptible to globbing

[This line](https://github.com/bats-core/bats-core/blob/118391d/libexec/bats-core/bats-exec-test#L73) is not a safe way to split arbitrary data into lines (as the suppressed [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206) notes), as it relies on Bash's wordsplitting logic which also expands globs.

MCVE:

```
$ bats --version
Bats 1.1.0

$ ls
bats-bug.bats  some-file

$ cat bats-bug.bats
#!/usr/bin/env bats

@test "bats bug" {
  run printf 'foo\n*\nbar\n'
  printf 'Output:\n%s\n\n' "$output"
  echo "Lines:"
  printf '  %s\n' "${lines[@]}"
  [[ "${lines[1]}" == "*" ]]
}

$ bats bats-bug.bats
 ✗ bats bug
   (in test file bats-bug.bats, line 7)
     `printf '  %s\n' "${lines[@]}"' failed
   Output:
   foo
   *
   bar

   Lines:
     foo
     bats-bug.bats
     some-file
     bar

1 test, 1 failure
```

I can send a PR to use `while read` instead, assuming there's no reason this behavior is intended (e.g. backwards compatibility).

--- a/lib/bats-core/test_functions.bash
+++ b/lib/bats-core/test_functions.bash
@@ -32,7 +32,7 @@ load() {
 run() {
   trap bats_interrupt_trap_in_run INT
   local origFlags="$-"
-  set +eET
+  set -f +eET
   local origIFS="$IFS"
   # 'output', 'status', 'lines' are global variables available to tests.
   # shellcheck disable=SC2034

--- a/test/bats.bats
+++ b/test/bats.bats
@@ -986,3 +986,18 @@ EOF
   [[ "${lines[5]}" == "# Received SIGINT, aborting ..." ]]
   [[ "${lines[6]}" == "# bats warning: Executed 2 instead of expected 1 tests" ]]
 }
+
+
+@test "single star in output is not treated as a glob" {
+  star(){ echo '*'; }
+  
+  run star
+  [ "${lines[0]}" = '*' ]
+}
+
+@test "multiple stars in output are not treated as a glob" {
+  stars(){ echo '**'; }
+  
+  run stars
+  [ "${lines[0]}" = '**' ]
+}

